### PR TITLE
Detect exclusive fullscreen on Windows

### PR DESCRIPTION
**Pull request by smoogipoo**
Prereqs:
- [x] https://github.com/ppy/osu-framework/pull/5204

### CODE REVIEW

File: osu.Game/Overlays/Settings/Sections/Graphics/LayoutSettings.cs [hunk 1]
```diff
@@ -213,6 +222,31 @@ void updateScalingModeVisibility()
...
+            {
+                windowModeDropdown.WarningText = GraphicsSettingsStrings.NotFullscreenNote;
+                return;
+            }
+
+            switch (fullscreenCapability.Value)
+            {
+                case FullscreenCapability.Unknown:
+                    if (host.Window is WindowsWindow)
+                        windowModeDropdown.WarningText = "Checking for exclusive fullscreen...";
```
Comment (peppy) [inline osu.Game/Overlays/Settings/Sections/Graphics/LayoutSettings.cs:237]:
```suggestion
                        windowModeDropdown.WarningText = "Checking fullscreen capabilities...";
```

File: osu.Game/Overlays/Settings/Sections/Graphics/LayoutSettings.cs [hunk 2]
```diff
@@ -213,6 +222,31 @@ void updateScalingModeVisibility()
...
+                    break;
+
+                case FullscreenCapability.Capable:
+                    windowModeDropdown.WarningText = default;
```
Comment (peppy) [inline osu.Game/Overlays/Settings/Sections/Graphics/LayoutSettings.cs:241]: Given that this text shows a message while "loading", I'd probably expect this to show a positive message afterwards. ie. "osu! is successfully running in exclusive fullscreen, guaranteeing low latency"

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game/Overlays/Settings/Sections/Graphics/LayoutSettings.cs (hunk 2 above)
Comment (peppy) [reply osu.Game/Overlays/Settings/Sections/Graphics/LayoutSettings.cs]: Yeah if it's involved it can come later, let's see how long it takes to annoy me.

**Comment by peppy** [comment]
This shows on macOS as supported. I guess that was the intention though?

**Comment by smoogipoo** [comment]
The `Capable` state is best effort I guess. Could hide it for Linux/macOS.

**Comment by peppy** [comment]
I've refactored the `WarningText` flow a touch to allow for different colouring. Let me know if you want it as a separate PR (was going to do this but it conflicts with the changes here so not sure it's worth the effort unless it will help with review).

osu_2022-05-31_14-05-54

**Review by peppy — DISMISSED** [review]

File: osu.Game/Overlays/Settings/SettingsItem.cs [hunk 3]
```diff
@@ -70,27 +70,32 @@ public virtual LocalisableString LabelText
...
-                    // construct lazily for cases where the label is not needed (may be provided by the Control).
-                    FlowContent.Add(warningText = new SettingsNoticeText(colours) { Margin = new MarginPadding { Bottom = 5 } });
-                }
+        /// <summary>
+        /// Set the text to be displayed at the bottom of this <see cref="SettingsItem{T}"/>.
+        /// Generally used to recommend the user change their setting as the current one is considered sub-optimal.
+        /// </summary>
+        /// <param name="text">The text to display.</param>
+        /// <param name="isWarning">Whether the text is in a warning state. Will decide how this is visually represented.</param>
+        public void SetWarningText(LocalisableString text, bool isWarning = true)
```
Comment (frenzibyte) [inline osu.Game/Overlays/Settings/SettingsItem.cs:87]: Having `warning` in the method name but allow setting a green note instead feels counter-intuitive, would rename it to something like `SetNote`/`SetNoteText` and default `isWarning` to false, or remove the default.

Comment (peppy) [reply osu.Game/Overlays/Settings/SettingsItem.cs]: Have applied this change.

**Comment by peppy** [comment]
@smoogipoo This one should be good to go, need your check on the changes I've made.

**Review by peppy — APPROVED** [review]

**Comment by smoogipoo** [comment]
Looks fine

**Pull request merged**